### PR TITLE
Allow helpers in response rule value

### PR DESCRIPTION
**Pull request by Jerantony**
**Issue**
https://github.com/mockoon/mockoon/issues/988

**Technical implementation details**

**Checklist**

- [ ] data migration added (@mockoon/commons)
- [ ] data migration automated tests added (@mockoon/commons)
- [ ] CLI automated tests added (@mockoon/cli)
- [ ] desktop automated tests added (@mockoon/desktop)

### CODE REVIEW

File: packages/commons-server/src/libs/response-rules-interpreter.ts [hunk 1]
```diff
@@ -198,4 +203,27 @@ export class ResponseRulesInterpreter {
...
+    try {
+      parsedValue = TemplateParser(
+        false,
+        value,
+        this.environment,
+        this.processedDatabuckets,
+        this.request
+      );
+    } catch (error) {
+      return value;
```
Comment (Jerantony) [inline packages/commons-server/src/libs/response-rules-interpreter.ts:224]: I thought it makes sense to simply use the unparsed value if parsing fails. Any better idea?

**Comment by 255kb** [comment]
Thank you for opening this PR, could you please comment on the #988 issue so I can assign it to you?

File: packages/commons-server/src/libs/response-rules-interpreter.ts (hunk 1 above)
Comment (255kb) [reply packages/commons-server/src/libs/response-rules-interpreter.ts:224]: No, this makes sense.

File: packages/commons-server/src/libs/response-rules-interpreter.ts [hunk 2]
```diff
@@ -175,7 +180,7 @@ export class ResponseRulesInterpreter {
       return value.includes(rule.value);
     }
 
-    return String(value) === String(rule.value);
+    return String(value) === String(this.parseValue(rule.value));
```
Comment (255kb) [inline packages/commons-server/src/libs/response-rules-interpreter.ts:183]: I think I would parse the `rule.value` earlier, like on line ~165, so the reex or array include would also check on the parsed value.

**Review by 255kb — CHANGES_REQUESTED** [review]
Everything looks good. Thank you for fixing the tests too! 

The only thing I would change, would be to parse the rule.Value a bit earlier so it is used for the other type of rules too.

File: packages/commons-server/src/libs/response-rules-interpreter.ts (hunk 2 above)
Comment (Jerantony) [reply packages/commons-server/src/libs/response-rules-interpreter.ts:183]: Oh yes, that's a good point!

**Review by 255kb — APPROVED** [review]

**Comment by 255kb** [comment]
It's merged and will be in the next release (probably around september). Thank you!

**Pull request merged**